webapp/components/path.js

```diff
@@ -57,20 +62,40 @@ const PathInfo = (superClass) => class extends superClass {
   }
 
   computePathIsASubfolder(path) {
-    return !this.computePathIsATestFile(path)
-      && path && path.split('/').filter(p => p).length > 0;
+    if (!path || this.computePathIsATestFile(path)) {
+      return false;
+    }
+    return new URL(path, window.location).pathname.split('/').filter(p => p).length > 0;
   }
 
   computePathIsATestFile(path) {
-    return /(\.(html|htm|py|svg|xhtml|xht|xml)(\?.*)?$)/.test(path);
+    return /(\.(html|htm|py|svg|xhtml|xht|xml)(\?.*)?$)/.test(new URL(path, window.location).pathname);
```

Comment:
It took me a while (after looking up MDN) to understand what this does -- getting rid of the query string in `path` IIUC. This might warrant a comment.